internal/logger/logger.go

```diff
@@ -2,32 +2,72 @@
 package logger
 
 import (
-	"log"
-
-	"go.uber.org/zap"
+	"io"
+	"log/slog"
+	"os"
 )
 
+// Logger is a wrapper around slog.Logger that provides additional methods
+type Logger struct {
+	*slog.Logger
+	closer io.Closer
+}
+
 // NewLogger builds the standard logger for terragrunt-ls.
 //
 // When supplied with a filename, it'll create a new file and write logs to it.
 // Otherwise, it'll write logs to stderr.
-func NewLogger(filename string) *zap.Logger {
-	if filename != "" {
-		config := zap.NewDevelopmentConfig()
-		config.OutputPaths = []string{filename}
-
-		logger, err := config.Build()
-		if err != nil {
-			log.Fatal(err)
-		}
+func NewLogger(filename string) *Logger {
+	if filename == "" {
+		handler := slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
```

Comment:
imho the default log format is better for human readability on stderr, while JSON makes sense for file output.